docker/Dockerfile

```diff
@@ -93,46 +90,67 @@ RUN python /usr/local/generate_toolchain.py --prefix /usr/local
+    -DLLVM_BINUTILS_INCDIR=/usr/include
+
+# SPIRV LLVM Translator
+RUN install-cmake-from-source https://github.com/KhronosGroup/SPIRV-LLVM-Translator/archive/v11.0.0.tar.gz \
+    -DBUILD_SHARED_LIBS=ON \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+    -DCMAKE_SKIP_RPATH=ON \
+    -DLLVM_INCLUDE_TESTS=ON \
+    -DLLVM_EXTERNAL_LIT=/usr/bin/lit
+
+# opencl-clang
+COPY usr/local/package/opencl-clang.sh /usr/local/package/opencl-clang.sh
+RUN /usr/local/package/opencl-clang.sh https://github.com/intel/opencl-clang/archive/v11.0.0.tar.gz
 
-# Libpci access and libdrm for communicating with graphics hardware
+# intel graphics compiler
+COPY usr/local/package/intel-graphics-compiler.sh /usr/local/package/intel-graphics-compiler.sh
+RUN /usr/local/package/intel-graphics-compiler.sh https://github.com/intel/intel-graphics-compiler/archive/igc-1.0.5435.tar.gz
+
+# intel compute runtime dependencies
+# gmmlib
+RUN install-cmake-from-source https://github.com/intel/gmmlib/archive/intel-gmmlib-20.4.1.tar.gz \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    -DCMAKE_INSTALL_LIBDIR=lib \
+    -DRUN_TEST_SUITE=OFF
+# libpciaccess and libdrm help communicate with graphics hardware
 RUN install-from-source https://xorg.freedesktop.org/releases/individual/lib/libpciaccess-0.16.tar.bz2
-RUN install-meson-from-source https://dri.freedesktop.org/libdrm/libdrm-2.4.100.tar.bz2 \
+RUN install-meson-from-source https://dri.freedesktop.org/libdrm/libdrm-2.4.104.tar.xz \
     -Dudev=false \
     -Dvalgrind=false
+# libva
+RUN install-from-source https://github.com/intel/libva/releases/download/2.10.0/libva-2.10.0.tar.bz2 \
+    -DENABLE_VA_MESSAGING=ON
 
-# Intel graphics compiler
-RUN install-cmake-from-source https://github.com/KhronosGroup/SPIRV-LLVM-Translator/archive/v9.0.0-1.tar.gz
-COPY usr/local/package/opencl-clang.sh /usr/local/package/opencl-clang.sh
-RUN /usr/local/package/opencl-clang.sh https://github.com/intel/opencl-clang/archive/v9.0.0.tar.gz
-RUN install-from-source https://github.com/intel/intel-graphics-compiler/archive/igc-1.0.3041.tar.gz \
-    -DIGC_OPTION__ARCHITECTURE_TARGET='Linux64' \
-    -DIGC_PREFERRED_LLVM_VERSION='9.0.1'
-RUN install-from-source https://github.com/intel/gmmlib/archive/intel-gmmlib-19.4.1.tar.gz \
-    -DRUN_TEST_SUITE=OFF
-
-# Libva and the iHD driver for hardware compression
-RUN install-from-source https://github.com/intel/libva/releases/download/2.6.0/libva-2.6.0.tar.bz2
-RUN install-from-source https://github.com/intel/media-driver/archive/intel-media-19.4.0r.tar.gz \
+# intel media driver
+RUN install-cmake-from-source https://github.com/intel/media-driver/archive/intel-media-20.4.5.tar.gz \
     -DINSTALL_DRIVER_SYSCONF=OFF \
     -DLIBVA_DRIVERS_PATH="/usr/local/lib/dri" \
     -DMEDIA_RUN_TEST_SUITE=OFF \
     -DBUILD_TESTING=OFF \
     -DBUILD_KERNELS=ON \
     -DENABLE_KERNELS=ON \
     -DENABLE_NONFREE_KERNELS=ON \
-    -DBUILD_TYPE=Release
+    -DCMAKE_INSTALL_PREFIX=/usr \
```

Comment:
Why does this have to go in `/usr` this will cause problems when you try to install onto a robot

Reply:
In this example I copied over the flags from the arch PKGBUILD verbatim. I can try it with `/usr/local` or something else if you like?

Reply:
Everything we build that needs to go on the robot should be in /usr/local

Reply:
Also, `install-from-source` already sets `CMAKE_INSTALL_PREFIX` for you

Reply:
`install-from-source` didn't set it for CMake builds, but I have added that because you make a good point that it really should.

Reply:
Sure it does, via the toolchain file https://github.com/NUbots/NUbots/blob/master/docker/usr/local/toolchain/generate_toolchains.py#L34

Reply:
My bad, the [toolchain file](https://github.com/NUbots/NUbots/blob/master/docker/usr/local/toolchain/generate_toolchains.py#L34) sets it

Reply:
ahh my bad too. I forgot about those args